src/SWP/Component/Common/Generator/GeneratorInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
```

Comment:
You can have also other generators like random number generator etc. then I see no problem with GeneratorInterface name.